Add docs on dumps (#21704)
dotnet/docs: docs/core/diagnostics/index.md

[Unit testing](../testing/index.md) is a key component of continuous integration and deployment of high-quality software. Unit tests are designed to give you an early warning when you break something.

## Dumps

> A [dump](./dumps.md) is a file that contains snapshot of the process at the time they were created and can be useful for examining the state of your application for debugging various issues.

Reviewer: ```suggestion
A [dump](./dumps.md) is a file that contains a snapshot of the process at the time of creation. These can be useful for examining the state of your application for debugging purposes.
```
